bundles/org.eclipse.equinox.app/osgi/org/osgi/service/application/ApplicationAdminPermission.java

```diff
@@ -325,13 +325,6 @@ private static class SignerWrapper extends Object {
 		private String pattern;
 		private ApplicationDescriptor appDesc;
 
-		/**
-		 * @param pattern
-		 */
-		public SignerWrapper(String pattern) {
```

Comment:
@jjohnstn can you have a look here? This seems to be removed as part of the removal of **private (!)** unused constructors cleanup ... it does not looks right!

Reply:
Thanks that makes sense, and is interesting that this can be even detected! I think I need to check if pattern is used at all then...

Reply:
@tjwatson I now further investigated here and the constructor is not used. That makes pattern never have any other value than `null` this results in being the equals method useless. I also only found one place where it is used that is here

https://github.com/eclipse-equinox/equinox/blob/2fd1eb7b348f3a3925b60f0c33dafeb6feae4b9a/bundles/org.eclipse.equinox.app/osgi/org/osgi/service/application/ApplicationAdminPermission.java#L212-L223

so it could be replaced with

`props.put( "signer", other.applicationDescriptor );`

I'm just not sure if this is useful for anything, can filter really compare arbitrary objects? Can you please have a look how to resolve this issue here?

Reply:
I get the feeling that the constructor is maybe needed to construct a value from the string in a filter @tjwatson is this correct? In this case it seems most useful to make it a public final internal class maybe to prevent it from being considered unused?

@jjohnstn As it is always the case that one can call a public constructor from reflection, should maybe the cleanup not remove the constructor even if it look unused?

Reply:
The easiest workaround for me seems to make this class package private (or move it into an internal package) @tjwatson WDYT?

Reply:
You are right... any reason we embedd the code here instead of consume it directly from the official jars via imports?